mlrun/runtimes/base.py

```diff
@@ -1144,6 +1150,18 @@ def verify_base_image(self):
             self.spec.build.base_image = image
             self.spec.image = ""
 
+    def verify_run_params(self, parameters: typing.Dict[str, typing.Any]):
+        for param_name, param_value in parameters.items():
+
+            if isinstance(param_value, dict):
+                self.verify_run_params(param_value)
+
```

Comment:
would add a comment as well that we are using recursion because of the possible nested params etc...